### PR TITLE
Feature/split adapters

### PR DESCRIPTION
**Pull request by diba1013**

### CODE REVIEW

**Comment by roesslerj** [comment]
This makes XmlInputHandler obsolete, doesn't it? Forgot to delete...

File: src/main/java/de/retest/recheck/xml/FileXmlRecheckAdapter.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	public Set<RootElement> convert( final Object toVerify ) {
+		final File file = (File) toVerify;
+		try ( final InputStream in = new FileInputStream( file ) ) {
+			return XmlToRootConverter.toRootElement( in );
+		} catch ( final IOException e ) {
+			logger.error( "Could not read file to xml '{}'.", file, e );
+		} catch ( final XMLStreamException e ) {
+			logger.error( "Failed to convert xml of file '{}'.", file, e );
+		}
+		return Collections.emptySet();
```
Comment (roesslerj) [inline src/main/java/de/retest/recheck/xml/FileXmlRecheckAdapter.java]: Why not throw exception?

File: src/main/java/de/retest/recheck/xml/StringXmlRecheckAdapter.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	@Override
+	public Set<RootElement> convert( final Object toVerify ) {
+		try ( final Reader reader = new StringReader( (String) toVerify ) ) {
+			return XmlToRootConverter.toRootElement( reader );
+		} catch ( final IOException e ) {
+			logger.error( "Could not read string '{}'.", toVerify, e );
+		} catch ( final XMLStreamException e ) {
+			logger.error( "Failed to convert xml '{}'.", toVerify, e );
+		}
+		return Collections.emptySet();
```
Comment (roesslerj) [inline src/main/java/de/retest/recheck/xml/StringXmlRecheckAdapter.java]: Same: Why not throw exception?

File: src/main/java/de/retest/recheck/xml/XmlToRootConverter.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		}
+		try {
+			while ( eventReader.hasNext() ) {
+				final XMLEvent event = eventReader.nextEvent();
+				if ( event.isStartElement() ) {
+					final XMLEventToRootElementConverter converter = new XMLEventToRootElementConverter( event );
+					return Collections.singleton( (RootElement) converter.convert( eventReader ) );
+				}
+			}
+			return Collections.emptySet();
```
Comment (roesslerj) [inline src/main/java/de/retest/recheck/xml/XmlToRootConverter.java]: Same: Why not throw exception?

**Review by roesslerj — CHANGES_REQUESTED** [review]

File: src/main/java/de/retest/recheck/xml/FileXmlRecheckAdapter.java (hunk 1 above)
Comment (diba1013) [reply src/main/java/de/retest/recheck/xml/FileXmlRecheckAdapter.java]: Is there no defined behaviour if a empty collection is returned?

Comment (roesslerj) [reply src/main/java/de/retest/recheck/xml/FileXmlRecheckAdapter.java]: `throw new IllegalStateException( "Cannot check empty state!" );` But then we don't know why...

**Review by roesslerj — APPROVED** [review]

**Pull request merged**